packages/cli/src/commands/site/deploy/index.ts

```diff
@@ -101,30 +101,24 @@ const command: CommandModule<
           );
         }
 
-        if (autoVersion != null && argv.autoVersion !== autoVersion.type) {
-          consola.debug(
-            `Overriding "autoVersion" from config file with ${argv.autoVersion}`,
-          );
-          if (argv.autoVersion !== "git-describe") {
-            throw new Error(
-              `Only 'git-describe' is supported for autoVersion`,
-            );
-          }
-        }
-
-        if (directory != null && argv.directory !== directory) {
-          consola.debug(
-            `Overriding "directory" from config file with ${argv.directory}`,
+        if (
+          autoVersion != null && argv.autoVersion !== autoVersion.type
+          && argv.autoVersion !== "git-describe"
```

Comment:
Minor note I feel this block and the one above are going to be difficult to maintain if we add another auto version type. The checks also seem slightly loose for example the above one will throw with `--autoVersion git-describe --tagPrefix abc` if there is no config file but that would be a valid state

Reply:
Yeah I agree, but it is required I think to not get the user in a state that doesn't make sense.

As for the part that didn't work, that was just because I forgot to check if autoVersion was defined.

Open to ideas here honestly to make this easier to handle.

Reply:
I would recommend breaking up the check a bit for example:

```
const autoVersionType = argv.autoVersion ?? autoVersion;
if (autoVersionType !== "git-describe") { throw }
```

To make it extensible in the future we may be able to do something with the AutoVersionType type and visit against each of the cases in the union and if we don't match any then we would throw